### PR TITLE
etcdserver: parallelize expired leases process

### PR DESCRIPTION
**Pull request by fanminshi**
When 1000 leases expired at the same time, etcd takes more than 5 seconds to clean them. This means that even after the leases have expired, keys associated with leases are still accessible. I increase the deletion throughput by parallelizing leases deletion process.

change variable name and comment for easier read. Fix a race condition in the parallelized leases revoked logic.

### CODE REVIEW

**Comment by fanminshi** [comment]
Move PR from https://github.com/coreos/etcd/pull/6459 to this one.

File: etcdserver/server.go [hunk 1]
```diff
@@ -618,9 +621,20 @@ func (s *EtcdServer) run() {
...
+
 			s.goAttach(func() {
-				for _, l := range leases {
-					s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(l.ID)})
+				// Increases throughput of expired leases deletion process through parallelization
+				c := make(chan struct{}, maxPendingRevokes)
+				for _, lease := range leases {
+					select {
+					case c <- struct{}{}:
+					case <-s.done:
```
Comment (heyitsanthony) [inline etcdserver/server.go]: `s.stopping`

File: etcdserver/server.go [hunk 2]
```diff
@@ -618,9 +621,20 @@ func (s *EtcdServer) run() {
...
+						return
+					}
+					go func(leaseID int64) {
```
Comment (heyitsanthony) [inline etcdserver/server.go]:
```go
lid := lease.ID
s.goAttach(func() {
    s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(lid)})
    <-c
})
```

File: etcdserver/server.go [hunk 3]
```diff
@@ -618,9 +621,20 @@ func (s *EtcdServer) run() {
 			f := func(context.Context) { s.applyAll(&ep, &ap) }
 			sched.Schedule(f)
 		case leases := <-expiredLeaseC:
+
```
Comment (heyitsanthony) [inline etcdserver/server.go]: remove superfluous space

**Comment by heyitsanthony** [comment]
lgtm following comment fixups

File: etcdserver/server.go (hunk 1 above)
Comment (fanminshi) [inline etcdserver/server.go]: done!

File: etcdserver/server.go (hunk 2 above)
Comment (fanminshi) [inline etcdserver/server.go]: done!

File: etcdserver/server.go (hunk 3 above)
Comment (fanminshi) [inline etcdserver/server.go]: done!

**Pull request merged**